de.dlr.sc.virsat.model.extension.cefx.ui/src/de/dlr/sc/virsat/model/extension/cefx/ui/importWizards/CometImportWizardPage.java

```diff
@@ -0,0 +1,180 @@
+/*******************************************************************************
+ * Copyright (c) 2008-2019 German Aerospace Center (DLR), Simulation and Software Technology, Germany.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *******************************************************************************/
+
+package de.dlr.sc.virsat.model.extension.cefx.ui.importWizards;
+
+import org.eclipse.jface.wizard.WizardPage;
+import org.eclipse.swt.SWT;
+import org.eclipse.swt.events.SelectionAdapter;
+import org.eclipse.swt.events.SelectionEvent;
+import org.eclipse.swt.layout.GridData;
+import org.eclipse.swt.layout.GridLayout;
+import org.eclipse.swt.widgets.Button;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.Label;
+import org.eclipse.swt.widgets.Text;
+import org.eclipse.swt.widgets.Tree;
+import org.eclipse.swt.widgets.TreeItem;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * The CometImportWizardPage class represents a wizard page used to configure 
+ * and fetch data from a COMET server. It provides a UI for entering server details, 
+ * initiating data fetch, and displaying the fetched data in a tree structure.
+ */
+public class CometImportWizardPage extends WizardPage {
+
+    private static final int GRID_COLUMNS = 2;
+    private static final int VERTICALSPAN = 1;
+    private static final int TREE_HEIGHT_HINT = 400;
+    private static final int TREE_WIDTH_HINT = 600;
+
+    private Text serverUrlText;
+    private Text loginText;
+    private Text passwordText;
+    private Tree targetTree;
+    private CometDataFetcher fetcher;
+
+    /**
+     * Constructs a new CometImportWizardPage with the specified name.
+     * 
+     * @param pageName The name of the wizard page.
+     */
+    protected CometImportWizardPage(String pageName) {
+        super(pageName);
+        setTitle("Comet Server Configuration");
+        setDescription("Enter details and fetch data.");
+    }
+
+    /**
+     * Creates the controls (UI elements) for this wizard page.
+     */
+    @Override
+    public void createControl(Composite parent) {
+        Composite container = new Composite(parent, SWT.NONE);
+        GridLayout layout = new GridLayout(GRID_COLUMNS, false);
+        container.setLayout(layout);
+
+        createLabelAndTextField(container, "Server URL:", SWT.BORDER);
+        createLabelAndTextField(container, "Login:", SWT.BORDER);
+        createLabelAndTextField(container, "Password:", SWT.BORDER | SWT.PASSWORD);
+
+        Button fetchButton = new Button(container, SWT.PUSH);
+        fetchButton.setText("Fetch Data");
+        fetchButton.addSelectionListener(new SelectionAdapter() {
```

Comment:
Could you maybe handle the add event in a dedicated function? There you could then use an eclipse job with progress monitor... So that users get feedback that something is being done in the background?

Maybe take some inspiration from here:  https://github.com/virtualsatellite/VirtualSatellite4-Core/blob/21c05a97ab73c79bf96aad96807fca7ebbbe0239/de.dlr.sc.virsat.model.extension.mechanical.ui/src/de/dlr/sc/virsat/model/extension/mechanical/ui/wizards/CadImportWizard.java#L87

Reply:
I've updated the code to ensure that UI elements are accessed only on the UI thread, using syncExec to safely retrieve input values before starting the job, which prevents the Invalid thread access error and implement the Visible Progress Dialog.